firedrake/mesh.py

```diff
@@ -437,8 +437,15 @@ def add_overlap():
         nfacets = self.comm.allreduce(nfacets, op=MPI.MAX)
 
         self._grown_halos = False
-        self._ufl_cell = ufl.Cell(_cells[dim][nfacets])
-
+        # Note that the geometric dimension of the cell is not set here
+        # despite it being a property of a UFL cell. It will default to
+        # equal the topological dimension.
+        # Firedrake mesh topologies, by convention, which specifically
+        # represent a mesh topology (as here) have geometric dimension
+        # equal their topological dimension. This is reflected in the
+        # corresponding UFL mesh.
+        cell = ufl.Cell(_cells[tdim][nfacets])
+        self._ufl_mesh = ufl.Mesh(ufl.VectorElement("Lagrange", cell, 1, dim=cell.topological_dimension()))
```

Comment:
Does this break the `point` cell case where you need `VectorElement("DG", cell, 0, ...)` ? Or will you do that in a later round?

Reply:
It seems to work fine with
```python
        cell = ufl.Cell("vertex")
        self._ufl_mesh = ufl.Mesh(ufl.VectorElement("DG", cell, 1, dim=cell.topological_dimension()))
```
in the `VertexOnlyMeshTopology` introduced in #1638 (which needs updating).

Reply:
What I meant was that you presumably still need to special-case `point` here (but you will do it in #1638, so that is fine).

Reply:
Well that's what the `VertexOnlyMeshTopology` does (similarly to and `ExtrudedMeshTopology`. Interestingly 
```python
        cell = ufl.Cell("vertex")
        self._ufl_mesh = ufl.Mesh(ufl.VectorElement("Lagrange", cell, 1, dim=cell.topological_dimension()))
```
works fine and lets you create function spaces but I think this is because UFL doesn't have any checks on what elements are valid. If you try to make that element in FIAT it will fail.